examples/reflection/convert.rs

```diff
@@ -0,0 +1,68 @@
+//! Illustrates how to do reflection with conversions
+
+use bevy::{prelude::*, reflect::DynamicStruct};
+
+fn main() {
+    App::new()
+        .register_type::<Foo>()
+        .add_systems(Startup, setup)
+        .run();
+}
+
+/// Deriving `Reflect` implements the relevant reflection traits. In this case, it implements the
+/// `Reflect` trait the `Struct` trait, and the `FromReflect` trait. `derive(Reflect)` assumes that
+/// all fields also implement Reflect.
+///
+/// All fields in a reflected item will need to be `Reflect` as well. You can specify additional
+/// types that will be accepted per field using `#[reflect(from(...))]`
```

Comment:
It's really important to note _when_ these conversions matter. Inserting/applying a field into a type won't automatically convert it (yet, but that's probably best to save for a future PR). So this only matters for `FromReflect`. And we should be careful to explicitly state that.

I only mention this because some users have expressed confusion with `#[reflect(default)]` in that it doesn't control deserialization but how `FromReflect` operates.

Reply:
Does the added line cover it?

Reply:
I added another line. Does that cover it?